packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -147,12 +153,15 @@ class _TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
   List<double> columnWidths;
   List<bool> rootsExpanded;
 
+  LinkedScrollControllerGroup linkedHorizontalScrollControllerGroup;
+
   /// The number of items to show when animating out the tree table.
   static const itemsToShowWhenAnimating = 50;
 
   @override
   void initState() {
     super.initState();
+    linkedHorizontalScrollControllerGroup = LinkedScrollControllerGroup();
```

Comment:
instead of creating a LinkedScrollControllerGroup object in both FlatTable and TreeTable, can we just create this in _Table? That would require making _Table a stateful widget but would also eliminate some code duplication and less piping everything through to _Table

Reply:
Good idea. The other challenge is the builders in FlatTable and TreeTable need to take the LinkedScrollController but I think it is more idiomatic for them to take that as a parameter. There probably should be a whole family of helper classes in our package that use builders that accept a linked scroll controller as that is the efficient way to support bidirectional scrolling.